cluster/src/test/scala/org/apache/spark/sql/execution/benchmark/TPCDSQuerySnappyBenchmark.scala

```diff
@@ -0,0 +1,154 @@
+/*
+ * Copyright (c) 2017 SnappyData, Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you
+ * may not use this file except in compliance with the License. You
+ * may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+ * implied. See the License for the specific language governing
+ * permissions and limitations under the License. See accompanying
+ * LICENSE file.
+ */
+
+package org.apache.spark.sql.execution.benchmark
+
+import java.io.{File, FileOutputStream, PrintStream}
+
+import org.apache.spark.sql._
+import org.apache.spark.sql.catalyst.TableIdentifier
+import org.apache.spark.sql.catalyst.analysis.UnresolvedRelation
+import org.apache.spark.sql.catalyst.expressions.SubqueryExpression
+import org.apache.spark.sql.catalyst.plans.logical.LogicalPlan
+import org.apache.spark.sql.catalyst.util.fileToString
+import org.apache.spark.sql.types.StructType
+import org.apache.spark.util.Benchmark
+import org.apache.spark.{SparkConf, SparkContext}
+
+import scala.collection.mutable.ArrayBuffer
+
+
+object TPCDSQuerySnappyBenchmark {
+
+  var spark: SparkSession = _
+  var snappy: SnappySession = _
+  var ds: DataFrame = _
+
+  val tables = Seq("catalog_page", "catalog_returns", "customer", "customer_address",
+    "customer_demographics", "date_dim", "household_demographics", "inventory", "item",
+    "promotion", "store", "store_returns", "catalog_sales", "web_sales", "store_sales",
+    "web_returns", "web_site", "reason", "call_center", "warehouse", "ship_mode", "income_band",
+    "time_dim", "web_page")
+
+  var snappyRS: FileOutputStream = new FileOutputStream(
+    new File(s"Snappy_Results.out"))
+  var sparkRS : FileOutputStream = new FileOutputStream(
+    new File(s"Spark_Results.out"))
+
+  var snappyPS: PrintStream = new PrintStream(snappyRS)
+  var sparkPS: PrintStream = new PrintStream(sparkRS)
+
+  def setupTables(dataLocation: String, isSnappy: Boolean): Map[String, Long] = {
+    val props = Map("BUCKETS" -> "7")
```

Comment:
Why BUCKETS value is 7 here? is it hardcoded and we need to parameterize the same?

Reply:
Suyog, Can you please fix this.

Reply:
This is used by unit tests only & sufficient for test data used. If someone wants to change it, can directly replace with other number before running the test.